### PR TITLE
Post

### PR DESCRIPTION
**Pull request by timktchan**
Server 
- View Posts 
- Add Post 
- One posts 
  Problem: Generally working on postman, but when i add post, the username and _id will have a new id instead of of the actual username 

Client
- Ajax 
  Try to do one but don't know how to reflect it on front-end. 
  Trying using "this" but not sure if I do it right too.

### CODE REVIEW

File: public/application.js [hunk 1]
```diff
@@ -19,10 +17,144 @@ $(document).ready(function(){
...
+    account.signIn(username,password);
+  });
+
+  $('#signOut').click(function(){
+    event.preventDefault();
+    console.log('hello signout');
+
+    account.signOut();
   });
+  // $('#signUpSubmit').click(function(){
```
Comment (f3r) [inline public/application.js:94]: ?

File: public/post-client.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+  Post.prototype.showAllPosts = function(){
+    $.ajax({
+      context: this,
+      type: 'GET',
+      url: '/posts',
+      success: function(response){
+        var html = '';
+
+        response.reverse().forEach(function (post) {
```
Comment (f3r) [inline public/post-client.js:17]: you can tell mongo to sort however you want: http://docs.mongodb.org/manual/reference/operator/meta/orderby/

File: public/post-client.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    event.preventDefault();
+      post.title = $('#addTitle').val();
+      post.content = $('#addContent').val();
+      post.category = $('#addCategory').val();
+      post.photo = $('#addPhoto').val();
+
+    post.addPost();
+  });
+});
+  // var getAllPosts = function(){
```
Comment (f3r) [inline public/post-client.js:78]: ???

File: routes/posts.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        var db = request.server.plugins['hapi-mongodb'].db;
+        var ObjectId = request.server.plugins['hapi-mongodb'].ObjectID; 
+        db.collection('posts').findOne({_id: ObjectId(post_id)}, function(err,post){
+          if (err) {return reply('Internal MongoDB error', err);}
+
+          reply(post);
+        });
+      }
+    },
+    // {
```
Comment (f3r) [inline routes/posts.js:86]: ??????

File: routes/users.js [hunk 5]
```diff
@@ -13,13 +13,14 @@ exports.register = function(server, options, next){
...
           // Get user input parameters (username, email, password)
-          var user = {
-            username: request.payload.user.username,
-            email:    request.payload.user.email,
-            password: request.payload.user.password,
-            bio: request.payload.user.bio,
-            profilepic: request.payload.user.profilepic
-          };
+          var user = request.payload.user;
+          // var user = {
```
Comment (f3r) [inline routes/users.js:17]: ????????

File: public/application.js (hunk 1 above)
Comment (timktchan) [inline public/application.js:94]: Intention is to get user's input once user fill signup info and clicks the
button "signUpSubmit"

Comment (f3r) [inline public/application.js:94]: no commented code in PR please

**Pull request closed without merging**